netbout-web/src/main/js/inbox.js

```diff
@@ -51,48 +51,61 @@ $(document).ready(
-                      unseen = parseInt($bout.find('unseen').text(), 10);
-                    appendix += [
-                      '<li class="bout" id="bout',
-                      $bout.find('number').text(),
-                      '"><h1 class="bout"><span class="num',
-                      (unread === 0 && unseen === 0 ? '' : ' unread'),
-                      '">#',
-                      $bout.find('number').text(),
-                      '</span><a class="title" href="',
-                      $bout.find('link[rel="open"]').attr('href'),
-                      '">',
-                      escapeHTML($bout.find('title').text()),
-                      '</a>',
-                      (unread === 0 ? '' : '<span class="unread">' + unread + '</span>'),
-                      '</h1><div class="friends">'
-                    ].join('');
-                    $bout.find('friend').each(
-                      function (idx, friend) {
-                        var $friend = $(friend), shift = 57 * idx;
-                        appendix += [
-                          '<div class="friend" style="left:',
-                          shift + 'px;"><img alt="',
-                          escapeHTML($friend.find('alias').text()),
-                          '" src="',
-                          $friend.find('link[rel="photo"]').attr('href'),
-                          '"/></div>'
-                        ].join('');
-                      }
-                    );
-                    appendix += '</div></li>';
-                    more = $bout.find('link[rel="more"]').attr('href');
+                if (boutsAsXml) {
+                  boutsAsXml.find('bout').each(
+                    function (idx, bout) {
+                      var $bout = $(bout),
+                          unread = parseInt($bout.find('unread').text(), 10),
+                          unseen = parseInt($bout.find('unseen').text(), 10);
+                      appendix += [
+                        '<li class="bout" id="bout',
+                        $bout.find('number').text(),
+                        '"><h1 class="bout"><span class="num',
+                        (unread === 0 && unseen === 0 ? '' : ' unread'),
+                        '">#',
+                        $bout.find('number').text(),
+                        '</span><a class="title" href="',
+                        $bout.find('link[rel="open"]').attr('href'),
+                        '">',
+                        escapeHTML($bout.find('title').text()),
+                        '</a>',
+                        (unread === 0 ? '' : '<span class="unread">' + unread + '</span>'),
+                        '</h1><div class="friends">'
+                      ].join('');
+                      $bout.find('friend').each(
+                        function (idx, friend) {
+                          var $friend = $(friend), shift = 57 * idx;
+                          appendix += [
```

Comment:
@erimerturk let's extract this code (up tp line 89) into a separate function with a good name